### ISSUE TITLE
Evaluate panic when parameters is null

### ISSUE THREAD

**Issue by RadhiFadlillah** [Knetic/govaluate#65]
Hi @Knetic, first of all thanks for this package. I found a bug where `Evaluate` panic when its parameter is `nil`, even though this bug is supposed to be fixed in #10. I'm using `gopkg.in/Knetic/govaluate.v3`.

Here is my code :

```go
func main() {
	expression, err := govaluate.NewEvaluableExpression("X > 1")
	if err != nil {
		log.Println("Failed creating expression:", err)
		return
	}

	result, err := expression.Evaluate(nil)
	if err != nil {
		log.Println("Failed evaluating expression:", err)
		return
	}

	fmt.Println(result)
}
```

And here is the output :

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x4c2aee]

goroutine 1 [running]:
gopkg.in/Knetic/govaluate%2ev3.makeParameterStage.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x5813c0, 0x7fb5c73c1000)
        $HOME/GoProjects/src/gopkg.in/Knetic/govaluate.v3/evaluationStage.go:217 +0x2e
gopkg.in/Knetic/govaluate%2ev3.EvaluableExpression.evaluateStage(0x503ab5, 0x22, 0x1, 0xc42006a060, 0x3, 0x4, 0xc42001c1e0, 0x4fe8f3, 0x5, 0xc42001c140, ...)
        $HOME/GoProjects/src/gopkg.in/Knetic/govaluate.v3/EvaluableExpression.go:228 +0x180
gopkg.in/Knetic/govaluate%2ev3.EvaluableExpression.evaluateStage(0x503ab5, 0x22, 0x1, 0xc42006a060, 0x3, 0x4, 0xc42001c1e0, 0x4fe8f3, 0x5, 0xc42001c1e0, ...)
        $HOME/GoProjects/src/gopkg.in/Knetic/govaluate.v3/EvaluableExpression.go:168 +0x803
gopkg.in/Knetic/govaluate%2ev3.EvaluableExpression.Eval(0x503ab5, 0x22, 0x1, 0xc42006a060, 0x3, 0x4, 0xc42001c1e0, 0x4fe8f3, 0x5, 0x0, ...)
        $HOME/GoProjects/src/gopkg.in/Knetic/govaluate.v3/EvaluableExpression.go:159 +0xa1
gopkg.in/Knetic/govaluate%2ev3.EvaluableExpression.Evaluate(0x503ab5, 0x22, 0x1, 0xc42006a060, 0x3, 0x4, 0xc42001c1e0, 0x4fe8f3, 0x5, 0x0, ...)
        $HOME/GoProjects/src/gopkg.in/Knetic/govaluate.v3/EvaluableExpression.go:134 +0xfb
main.main()
        $HOME/GoProjects/src/test/main.go:16 +0x17c
exit status 2
```
If I put an empty `map` in `Evaluate`, the code will work as expected :

```go
...
        params := make(map[string]interface{})
	result, err := expression.Evaluate(params)
	if err != nil {
		log.Println("Failed evaluating expression:", err)
		return
	}
...
```
```
2017/07/23 14:51:00 Failed evaluating expression: No parameter 'X' found.
```

The weird thing is your [`TestNilParameterUsage`](https://github.com/Knetic/govaluate/blob/91facc4ae520fef82c9aee6b6ae720d9ae789131/evaluationFailure_test.go#L90]) is passed succesfully without problem. After looking around, I found that in your test the parameter is never `nil` because `runEvaluationFailureTests` will automatically [replace it](https://github.com/Knetic/govaluate/blob/91facc4ae520fef82c9aee6b6ae720d9ae789131/evaluationFailure_test.go#L523) with `EVALUATION_FAILURE_PARAMETERS`.

**Comment by Knetic** [comment]
Hey, good find! Looks like I outsmarted myself with that test. Also looks like that bug _used_ to be fixed, but there was a regression last summer which removed the line that turned nil parameters into `DUMMY_PARAMETERS` (an empty map). But because the test I wrote was flawed, nobody knew it was a problem.

Anyway, it should be fixed as of beae3c74e5d7b39c0674bc525b5fe6fd0167e47d. Thanks for the report and investigation!
